tests/Faker.Tests/Common/ResourceCollectionCachingTests.cs

```diff
@@ -0,0 +1,180 @@
+			{
+				var firstNames = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+				var lastNames = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.Last));
+
+				Assert.IsFalse(firstNames.SequenceEqual(lastNames));
+			}
+		}
+
+		[Test]
+		public void Switch_To_Foreign_Culture_Returns_Not_Equal_Arrays()
+		{
+			// Cache miss case
+			{
+				System.Threading.Thread.CurrentThread.CurrentUICulture = defaultUICulture;
+				var array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+				System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("de-DE");
+				var de_DE_array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+
+				Assert.IsFalse(array.SequenceEqual(de_DE_array));
+			}
+
+			// Cache hit case
+			{
+				System.Threading.Thread.CurrentThread.CurrentUICulture = defaultUICulture;
+				var array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+				System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("de-DE");
+				var de_DE_array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+
+				Assert.IsFalse(array.SequenceEqual(de_DE_array));
+			}
+		}
+
+		[Test]
+		public void Switch_Between_Foreign_Cultures_Returns_Not_Equal_Arrays()
+		{
+			// Cache miss case
+			{
+				System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("pt-BR");
+				var en_US_array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+				System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("de-DE");
+				var de_DE_array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+
+				Assert.IsFalse(en_US_array.SequenceEqual(de_DE_array));
+			}
+
+			// Cache hit case
+			{
+				System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("pt-BR");
+				var en_US_array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+				System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("de-DE");
+				var de_DE_array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+
+				Assert.IsFalse(en_US_array.SequenceEqual(de_DE_array));
+			}
+		}
+
+		[Test]
+		public void Switch_To_Unsupported_Culture_Returns_Equal_Arrays()
+		{
+			// Cache miss case
+			{
+				System.Threading.Thread.CurrentThread.CurrentUICulture = defaultUICulture;
+				var array = ResourceCollectionCacher.GetArray(PropertyHelper.GetProperty(() => Resources.Name.First));
+				System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("br-BR");
```

Comment:
Even if the test is for unsupported culture, you can not use a culture which isn't supported by the .NET framework. This only throws out an exception in the unit tests.